packages/projection/test/operators/Mappers/withNftMetadata.test.ts

```diff
@@ -0,0 +1,291 @@
+/* eslint-disable sonarjs/no-duplicate-string */
+import { Cardano, ChainSyncEventType, ChainSyncRollForward } from '@cardano-sdk/core';
+import { ChainSyncDataSet, SerializedChainSyncEvent, chainSyncData } from '@cardano-sdk/util-dev';
+import { Mappers, ProjectionEvent } from '../../../src';
+import { Observable, firstValueFrom, map, of } from 'rxjs';
+import { dummyLogger } from 'ts-log';
+
+// Events in inline-datum.json have both cip25 and cip68 metadata.
+// Removing cip25 metadata helps testing cip68.
+const removeTxMetadata =
+  <T>() =>
+  (evt$: Observable<ProjectionEvent<T>>): Observable<ProjectionEvent<T>> =>
+    evt$.pipe(
+      map((evt) => ({
+        ...evt,
+        block: {
+          ...evt.block,
+          body: evt.block.body.map((tx) => ({
+            ...tx,
+            auxiliaryData: undefined
+          }))
+        }
+      }))
+    );
+
+const isDatumInBlock = (data: Cardano.Block) => {
+  if (!data || !data.body || !Array.isArray(data.body)) {
+    return false; // Invalid data format
+  }
```

Comment:
Is there any reason to check this? The type specifies that those properties are required
```suggestion
```

Reply:
This checks that the block has the proper structure before checking a datum. Removed on b37531e